src/component/common/ConstraintAccordion/ConstraintAccordionEdit/ConstraintAccordionEditBody/SingleLegalValue/SingleLegalValue.tsx

```diff
@@ -36,7 +36,15 @@ export const SingleLegalValue = ({
             <ConstraintFormHeader>
                 Add a single {type.toLowerCase()} value
             </ConstraintFormHeader>
-            <ConstraintValueSearch filter={filter} setFilter={setFilter} />
+            <ConditionallyRender
+                condition={Boolean(legalValues.length > 100)}
```

Comment:
The expression here is guaranteed to be a boolean, so we don't need to explicitly cast.